pkg/kubelet/kubelet_pods.go

```diff
@@ -550,7 +550,7 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 	// To avoid this users can: (1) wait between starting a service and starting; or (2) detect
 	// missing service env var and exit and be restarted; or (3) use DNS instead of env vars
 	// and keep trying to resolve the DNS name of the service (recommended).
-	serviceEnv, err := kl.getServiceEnvVarMap(pod.Namespace, *pod.Spec.EnableServiceLinks)
+	serviceEnv, err := kl.getServiceEnvVarMap(pod.Namespace, v1helper.GetEnableServiceLinks(&pod.Spec))
```

Comment:
/hold

actually, this was recently addressed in this PR:
https://github.com/kubernetes/kubernetes/pull/69061

the EnableServiceLinks value should be populated by default, so the helper is not really needed.

are you hitting this with a 1.12 kubelet and a 1.12 API server?
are they both the same MINOR version?

Reply:
I tried HEAD (v1.13.0) I pulled it last Friday, but forgot to update apiserver after build. So when I updated apiserver, poiter to EnableServiceLink became ok.
I'll close this PR and I'll look forward for your PR, in any case code should be robustness.